File: problem0001.js
/*
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
*/

function App(){
	var self = this;
	self.sum = 0;

	self.init = function(){
		self.instructions();
		self.getSum();
		self.printSum();
	};

	self.instructions = function(){
		console.log('If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.');
		console.log('Find the sum of all the multiples of 3 or 5 below 1000.');
	};

	self.getSum = function(){
		for(var i = 0; i < 1000; i++){
			if(i % 3 === 0 || i % 5 === 0){
				self.sum += i;
			}
		}
	}

	self.printSum = function(){
		console.log(self.sum);
	}
};

var a = new App();
a.init();

File: problem0001.go
package main

import "fmt"

func printInstructions() {
  fmt.Println("If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.")
  fmt.Println("Find the sum of all the multiples of 3 or 5 below 1000.")
}

func printAnswer(ans int) {
  fmt.Println("The answer is:", ans)
}

func sumMultiples(stopAt int) int{
  total := 0

  for i := 0; i < stopAt; i++ {
    if (i % 3 == 0) || (i % 5 == 0) {
      total += i
    }
  }

  return total
}

func main() {
  printInstructions()
  ans := sumMultiples(1000)
  printAnswer(ans)
}
